Source/Parsing/Frame/Content/StringParser/ID3FrameBooleanContentParser.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  ID3FrameBooleanContentParser.swift
+//
+//  Created by Nolaine Crusher on 03/30/2020.
+//  2018 Fabrizio Duroni.
+//
+
+import Foundation
+
+class ID3FrameBooleanContentParser {
+    private let paddingRemover: PaddingRemover
+    private let id3FrameConfiguration: ID3FrameConfiguration
+    
+    init(paddingRemover: PaddingRemover,
+         id3FrameConfiguration: ID3FrameConfiguration) {
+        self.paddingRemover = paddingRemover
+        self.id3FrameConfiguration = id3FrameConfiguration
+    }
+    
+    func parse(frame: Data, version: ID3Version) -> Bool? {
+        let headerSize = id3FrameConfiguration.headerSizeFor(version: version)
+        let frameContentRangeStart = headerSize
+        let frameContent = frame.subdata(in: frameContentRangeStart..<frame.count)
+        if let frameContentAsString = String(data: frameContent, encoding: .utf8).flatMap(Bool.init) {
```

Comment:
Should this be `utf8` or another encoding? This is the example I found for how to convert `Data` to `Bool` but I'm not sure if there is another way.

Reply:
That doesn’t look right. That would turn `[0x00, 0x00, 0x00, 0x01]` into `"[null][null][null][private use one]"` and then feed it to the [`Bool` initializer](https://developer.apple.com/documentation/swift/bool/2428231-init) expecting `"true"` or `"false"`.

You just need to pull the relevant byte (a `UInt8`) and see if it is equal to `0`.

Reply:
That...was silly of me. I struggled with that forever last night.

So I just need:

```swift
if frameContent (converted to UInt8, if it's not already).last == 0x01 {
return true
} else {
return false
}
```

Once again overthinking things has made me seem ridiculous.

Reply:
Yes. It is probably better to test for `0` though and let everything else be `true`. In the case of some other tool misbehaving, an undefined `2` is more likely to intend the same as `1` than the same as `0`.

Reply:
Good point. I'll fix that.